lib/Test/Nginx/Util.pm

```diff
@@ -1238,32 +1238,42 @@ sub parse_headers ($) {
     return \%headers;
 }
 
-sub expand_env_in_text ($$) {
-    my ($text, $name) = @_;
+sub expand_env_in_text ($$$) {
+    my ($text, $name, $rand_ports) = @_;
 
     if (!defined $text) {
         return;
     }
 
-    my $used_ports = { $ServerPort => 1 };
+    my $used_ports = {
+        $ServerPort => 1,
+        map { $_ => 1 } values %$rand_ports
+    };
 
     $text =~ s/\$(TEST_NGINX_[_A-Z0-9]+)/
+        my $repl;
         if ($1 =~ m{^(TEST_NGINX_RAND_PORT_[0-9]+)$}) {
-            if (!defined $ENV{$1}) {
+            if (!defined $rand_ports->{$1}) {
                 my $rand_port = gen_rand_port 1000, $used_ports;
 
                 if (!defined $rand_port) {
                     bail_out "$name - Cannot find an available listening port number for $1.\n";
                 }
 
-                $ENV{$1} = $rand_port;
+                $rand_ports->{$1} = $rand_port;
                 $used_ports->{$rand_port} = 1;
-            }
+                $repl = $rand_port;
 
-        } elsif (!defined $ENV{$1}) {
-            bail_out "$name - No environment $1 defined.\n";
+            } else {
+                $repl = $rand_ports->{$1};
+            }
+        } else {
+            if (!defined $ENV{$1}) {
+                bail_out "$name - No environment $1 defined.\n";
+            }
+            $repl = $ENV{$1};
         }
-        $ENV{$1}/eg;
+        $repl/eg;
```

Comment:
`/eg` should be on a separate line and with one-level-less indentation.

Reply:
Okay, it's been fixed.